### PR TITLE
Add txs_commitment to Da Header

### PR DESCRIPTION
**Pull request by kpp**
# Description
https://github.com/chainwayxyz/citrea/pull/229 introduced a system contract which requires a merkle root to be passed to L1BlockHashList::setBlockInfo. This PR introduces a txs_commitment in BlockHeader to be passed to Evm::begin_soft_confirmation_hook.

## Linked Issues
- Related to https://github.com/chainwayxyz/citrea/issues/228

## Testing
`make test`

### CODE REVIEW

File: crates/sovereign-sdk/rollup-interface/src/state_machine/da.rs [hunk 1]
```diff
@@ -223,6 +223,9 @@ pub trait BlockHeaderTrait: PartialEq + Debug + Clone + Serialize + DeserializeO
     /// Hash the type to get the digest.
     fn hash(&self) -> Self::Hash;
 
+    /// Merkle root of the block.
+    fn merkle_root(&self) -> Self::Hash;
```
Comment (eyusufatik) [inline crates/sovereign-sdk/rollup-interface/src/state_machine/da.rs:227]: let's call this transaction_commitment everywhere, some da layers might not use merkle roots

File: crates/sovereign-sdk/module-system/sov-modules-stf-blueprint/src/lib.rs [hunk 2]
```diff
@@ -576,6 +576,7 @@ fn verify_soft_batch_signature<C: Context>(
     let unsigned = UnsignedSoftConfirmationBatch::new(
         soft_batch.da_slot_height(),
         soft_batch.da_slot_hash(),
+        soft_batch.da_slot_merkle_root(),
         soft_batch.pre_state_root(),
         soft_batch.txs(),
         soft_batch.l1_fee_rate(),
```
Comment (eyusufatik) [inline crates/sovereign-sdk/module-system/sov-modules-stf-blueprint/src/lib.rs:582]: in this file 

in begin_soft_batch function, observe how we assert values in soft confirmation against the ones full node itself supplied. 

da_slot_transaction_commitment should also be asserted that way.

File: crates/sovereign-sdk/adapters/celestia/src/celestia.rs [hunk 3]
```diff
@@ -240,6 +244,10 @@ impl BlockHeader for CelestiaHeader {
         TmHash(self.header.hash())
     }
 
+    fn merkle_root(&self) -> Self::Hash {
+        TmHash(self.header.merkle_root())
+    }
```
Comment (eyusufatik) [inline crates/sovereign-sdk/adapters/celestia/src/celestia.rs:249]: make unimplemented!() 

we don't need to update other adapters.

might be misleading for people, I don't know how celestia transaction commitments in the header work thb :)

File: crates/sovereign-sdk/adapters/avail/src/spec/header.rs [hunk 4]
```diff
@@ -53,6 +53,10 @@ impl BlockHeaderTrait for AvailHeader {
         self.hash.clone()
     }
 
+    fn merkle_root(&self) -> Self::Hash {
+        AvailHash::new(self.header.state_root)
+    }
```
Comment (eyusufatik) [inline crates/sovereign-sdk/adapters/avail/src/spec/header.rs:58]: unimplemented!()

File: crates/bitcoin-da/src/spec/header.rs [hunk 5]
```diff
@@ -25,6 +25,10 @@ impl BlockHeaderTrait for HeaderWrapper {
         BlockHashWrapper(self.header.block_hash())
     }
 
+    fn merkle_root(&self) -> Self::Hash {
+        BlockHashWrapper(BlockHash::from_raw_hash(self.header.merkle_root.into()))
+    }
+
```
Comment (eyusufatik) [inline crates/bitcoin-da/src/spec/header.rs:31]: we need serve the witness commitment here (post seg-wit update merkle root)

or else it'd be impossible on the rollup to use post seg-wit transactions

in service.rs we extract that for InclusionMultiProof

File: crates/evm/src/tests/hooks_tests.rs [hunk 6]
```diff
@@ -47,7 +53,8 @@ fn end_soft_confirmation_hook_sets_head() {
 
     evm.begin_soft_confirmation_hook(
         DA_ROOT_HASH.0,
-        GENESIS_STATE_ROOT.as_ref(),
+        pre_state_root,
+        &pre_state_root,
```
Comment (eyusufatik) [inline crates/evm/src/tests/hooks_tests.rs:57]: please define a constant called `DA_TX_COMMITMENT` and use that 
`pre_state_root` for both args might be confusing for readers

File: crates/evm/src/hooks.rs [hunk 7]
```diff
@@ -11,10 +11,11 @@ impl<C: sov_modules_api::Context> Evm<C>
 where
     <C::Storage as Storage>::Root: Into<[u8; 32]>,
 {
-    /// Logic executed at the beginning of the slot. Here we set the root hash of the previous head.
+    /// Logic executed at the beginning of the slot. Here we set the state root of the previous head.
     pub fn begin_soft_confirmation_hook(
         &self,
-        da_root_hash: [u8; 32],
+        da_slot_hash: [u8; 32],
+        _da_slot_merkle_root: [u8; 32],
```
Comment (eyusufatik) [inline crates/evm/src/hooks.rs:18]: good update with namings 👍🏻

**Review by eyusufatik — CHANGES_REQUESTED** [review]
Two main takeaways:
- Name it something like transaction_commitment
- For bitcoin da transaction commitment should be the seg-wit one not the original merkle root. service.rs and verifier.rs has examples

File: crates/sovereign-sdk/adapters/celestia/src/celestia.rs (hunk 3 above)
Comment (kpp) [reply crates/sovereign-sdk/adapters/celestia/src/celestia.rs]: https://docs.rs/celestia-tendermint/latest/celestia_tendermint/block/header/struct.Header.html#structfield.data_hash

**Review by eyusufatik — APPROVED** [review]

File: crates/bitcoin-da/src/spec/header.rs (hunk 5 above)
Comment (eyusufatik) [reply crates/bitcoin-da/src/spec/header.rs]: decided to do it in another issue

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**